src/server/config.js

```diff
@@ -1,48 +1,47 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
-var config = {};
+/**
+ * The Configuration model manages the configuration properties for the simulation.
+ * These properties are mapped with the options provided for the simulation, and
+ * are cross to any other instance involved in the simulation.
+ * @constructor
+ */
+function Configuration() {
+    this._config = {};
+}
 
-// module properties
 [
     { name: 'debugHostHandlers', optional: true },
     { name: 'forcePrepare' },
     { name: 'liveReload' },
-    { name: 'platform' },
-    { name: 'platformRoot' },
-    { name: 'projectRoot', single: true },
-    { name: 'server', optional: true },
     { name: 'simHostOptions' },
     { name: 'simulationFilePath' },
     { name: 'telemetry', optional: true },
     { name: 'touchEvents', optional: true },
     { name: 'xhrProxy', optional: true }
 ].forEach(function (prop) {
-    Object.defineProperty(module.exports, prop.name, {
+    Object.defineProperty(Configuration.prototype, prop.name, {
         get: function () {
-            return getValue(prop.name, prop.optional);
+            return getValue(this, prop.name, prop.optional);
         },
         set: function (value) {
-            setValue(prop.name, value, prop.single);
+            setValue(this, prop.name, value, prop.single);
         }
     });
 });
 
-function setValue(prop, value, single) {
-    if (single && config.hasOwnProperty(prop)) {
+function setValue(instance, prop, value, single) {
+    if (single && instance._config.hasOwnProperty(prop)) {
```

Comment:
If the property has been defined previously, then re-configuring it (e.g. specifying another setter) will throw `TypeError` before you get a chance to execute this code. I would instead recommend a more "javascriptish" way of applying `setValue()` function to the context of the call (and not passing `this` explicitly):

```javascript
Object.defineProperty(Configuration.prototype, prop.name, {
    get: function () {
        return getValue.apply(this, [prop.name, prop.optional]);
    },
    set: function (value) {
        return setValue.apply(this, [prop.name, value, prop.single]);
    }
}
```